agent-core/src/main/java/com/newrelic/agent/android/harvest/Harvest.java

```diff
@@ -73,21 +73,20 @@ public static void stop() {
         }
     }
 
-    public static void harvestNow(boolean finalizeSession) {
-        if (!isInitialized())
-            return;
-
-        if (finalizeSession) {
-            // add session meta to next harvest
-            instance.finalizeSession();
+    public static void harvestNow(boolean finalizeSession, boolean bWait) {
```

Comment:
This new method encapsulated changes to HarvestTimer to _restart the session_, which is to say resetting the the harvest times. Session times (total, time since start) are derived from this clock. It has not been hooked up to exiting code.